features/filesystem/fat/FATFileSystem.cpp

```diff
@@ -123,6 +123,61 @@ void fat_filesystem_set_errno(FRESULT res)
     return;
 }
 
+// Helper class for deferring operations when variable falls out of scope
+template <typename T>
+class Deferred {
+public:
+    T _t;
+    Callback<void(T)> _ondefer;
+
+    Deferred(const Deferred&);
+    Deferred &operator=(const Deferred&);
+
+public:
+    Deferred(T t, Callback<void(T)> ondefer = NULL)
+        : _t(t), _ondefer(ondefer)
+    {
+    }
+
+    operator T()
+    {
+        return _t;
+    }
+
+    ~Deferred()
+    {
+        if (_ondefer) {
+            _ondefer(_t);
+        }
+    }
+};
+
+static void dodelete(const char *data)
+{
+    delete[] data;
+}
+
+// Adds prefix needed internally by fatfs, this can be avoided for the first fatfs
+// (id 0) otherwise a prefix of "id:/" is inserted in front of the string.
+static Deferred<const char*> fat_path_prefix(int id, const char *path)
```

Comment:
Actually, reading tha GPT article, the format supports >128 partitions. The maximum should be 4294967295